code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -87,20 +87,17 @@
 		/datum/reagent/toxin/histamine,
 		/datum/reagent/medicine/morphine
 	)
+	var/list/recording_recipe
 
 	var/list/saved_recipes = list()
 
 /obj/machinery/chem_dispenser/Initialize()
 	. = ..()
+	dispensable_reagents = sortList(dispensable_reagents, /proc/cmp_reagents_asc)
 	if(emagged_reagents)
 		emagged_reagents = sortList(emagged_reagents, /proc/cmp_reagents_asc)
 	if(upgrade_reagents)
 		upgrade_reagents = sortList(upgrade_reagents, /proc/cmp_reagents_asc)
-	if(upgrade_reagents2)
```

Comment:
You sure about that? From looking elsewhere in the code, I think this means that you get NO upgraded reagents until you've FULLY upgraded your chem dispenser, which is kinda weird.

Reply:
Whoops, seems i fucked that up.